src/Subscriber/CheckoutPageSubscriber.php

```diff
@@ -375,4 +272,54 @@ public function getActiveToken($customField)
 
         return $activeToken;
     }
+    
+        
+    /**
+     * getApms
+     *
+     * @param  mixed $ckoContext
+     * @return void
+     */
+    public static function getApms($ckoContext)
+    {
+        $apmList = null;
+
+        if (array_key_exists("apms", $ckoContext) && count($ckoContext['apms']) > 0) {
+            $apmArray = $ckoContext['apms'];
+            foreach ($apmArray as $apm) {
+                $apmList[] = $apm['name'];
+
+                if ($apm['name'] == 'klarna') {
```

Comment:
How is this klarna specific? Do we really need reference at the top-level?
Why not referencing as `$apm['metadata']['details']['client_token'];`

Reply:
Why are we remapping to store plugin-specific attributes, for example `paymentMethodAvailable` ?